src/plugins/intel_gpu/src/runtime/kernels_cache.cpp

```diff
@@ -345,11 +347,13 @@ std::vector<kernel::ptr> kernels_cache::get_kernels(kernel_impl_params params) c
     }
     auto res = _kernels.find(params);
     OPENVINO_ASSERT(_kernels.end() != res, "Kernel for {" + current_node_id + "} is not found in the kernel cache!");
+    OPENVINO_ASSERT(res->second.size() != 0, "Number of kernels should not be zero for " + current_node_id);
 
-    std::vector<kernel::ptr> kernels;
-    kernels.reserve(res->second.size());
+    std::vector<kernel::ptr> kernels(res->second.size());
     for (auto& k : res->second) {
-        kernels.emplace_back(k->clone());
+        auto& kernel_ptr = k.first;
+        auto kernel_part_idx = k.second;
+        kernels[kernel_part_idx] = kernel_ptr->clone();
```

Comment:
How about do this in set_kernels ? 
We just can change the parameter type of the set_kernels function.

Reply:
Fixed it